internal/pkg/pct/install.go

```diff
@@ -111,3 +138,35 @@ func (p *PctInstaller) setupTemplateNamespace(targetDir string, info PuppetConte
 
 	return namespacePath, err
 }
+
+func (p *PctInstaller) downloadTemplate(targetURL *url.URL, downloadDir string) (string, error) {
+	// Get the file contents from URL
+	response, err := p.HTTPClient.Get(targetURL.String())
+	if err != nil {
+		return "", err
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode != 200 {
+		message := fmt.Sprintf("Received response code %d when trying to download from %s", response.StatusCode, targetURL.String())
+		return "", errors.New(message)
+	}
+
+	// Create the empty file
+	segments := strings.Split(targetURL.Path, "/")
+	fileName := segments[len(segments)-1]
```

Comment:
you could also use: `fileName := path.Base(response.Request.URL.Path)` It probably does the same sort of thing, but you never know with URLs...

Reply:
Good shout, that "just worked" 😊